cpp/src/io/parquet/reader_impl.cu

```diff
@@ -629,6 +635,26 @@ class aggregate_metadata {
         if (schema_elem.num_children == 0) {
           input_column_info& input_col =
             input_columns.emplace_back(input_column_info{schema_idx, schema_elem.name});
+
+          // set up child output column for one-level encoding list
+          if (schema_elem.is_one_level_list()) {
+            // determine the element data type
+            auto const element_type = to_type_id(
+              schema_elem, strings_to_categorical, timestamp_type_id, strict_decimal_types);
+            auto const element_dtype =
+              element_type == type_id::DECIMAL32 || element_type == type_id::DECIMAL64
+                ? data_type{element_type, numeric::scale_type{-schema_elem.decimal_scale}}
+                : data_type{element_type};
+
+            column_buffer element_col(element_dtype, schema_elem.repetition_type == OPTIONAL);
+            // store the index of this element
+            nesting.push_back(static_cast<int>(output_col.children.size()));
+            // TODO: not sure if we should assign a name or leave it blank
```

Comment:
Is the included in the output schema?
In ORC the names of nested columns are generated as the index in the parent's list of children. Gives a uniform way to access nested columns of lists/maps/structs. I don't know enough about Parquet to understand if the same logic can apply here.